app/controllers/internal_api/v1/pillars_controller.rb

```diff
@@ -18,4 +18,23 @@ def pillar_contents
       end
     end.deep_symbolize_keys
   end
+
+  # extra non pillar structures we need to have visible in salt
+  def non_pillar_contents
+    registry_contents
+  end
+
+  def registry_contents
+    {
+      registries: [].tap do |res|
+        DockerRegistry.all.each do |reg|
+          res << {
+            url:         reg.url,
+            mirror:      reg.mirror,
+            certificate: Certificate.find(reg.certifiable_id).certificate
```

Comment:
`certificate: reg.certificate.certificate` should work in order to serve the certificate for this registry.